lstgen/generators/go.py

```diff
@@ -0,0 +1,268 @@
+"""
+Go generator
+"""
+import ast
+from .. import (
+    parse_eval_stmt,
+    prepare_expr
+)
+from .. import (
+    EvalStmt,
+    IfStmt,
+    ThenStmt,
+    ElseStmt,
+    ExecuteStmt,
+)
+from .base import JavaLikeGenerator
+
+class GoGenerator(JavaLikeGenerator):
+    """ Java Generator """
+    bd_class = 'decimal.Decimal'
+    bd_class_constructor = 'decimal.NewFromInt'
+    list_const_parens = ('{', '}')
+    stmt_separator = ''
+    allow_constants = False
+
+    def __init__(self, parser, outfile, class_name=None, indent=None, package_name='default'):
+        super(GoGenerator, self).__init__(parser, outfile, class_name, indent)
+        self.package_name = package_name
+
+    def generate(self):
+        wr = self.writer
+        self._write_comment("This file is automatically generated by LstGen, do not edit!", False)
+        wr.writeln('package {}'.format(self.package_name or 'tax'))
+        wr.nl()
+        wr.writeln('import ( "github.com/shopspring/decimal" )')
+        wr.nl()
+
+        # Define the model as a struct.
+        with self.writer.indent('type {} struct'.format(self.class_name)):
+            wr.writeln("// ------------------------ Constants -------------------------")
+            for const in self.parser.constants:
+                if const.comment is not None:
+                    wr.nl()
+                    self._write_comment(const.comment, False)
+
+                if const.type.endswith('[]'):
+                    const.value = '[{}]'.format(const.value[1:-1])
+
+                thetype = self._convert_vartype(const.type)
+                wr.writeln('{const.name} {thetype}'.format(
+                    const=const,
+                    thetype=thetype
+                ))
+
+            for (comment, variables) in [
+                    ('Input variables', self.parser.input_vars),
+                    ('Output variables', self.parser.output_vars),
+                    ('Internal variables', self.parser.internal_vars),
+                ]:
+                wr.nl()
+                wr.writeln('// {}'.format(comment))
+
+                wr.writeln("// ------------------------ Variables -------------------------")
+                for var in variables:
+                    if var.comment is not None:
+                        wr.nl()
+                        self._write_comment(var.comment, False)
+                    vartype = self._convert_vartype(var.type)
+                    wr.writeln('{} {}'.format(var.name, vartype))
+
+        # Define the New() method that also sets default values.
+        with self.writer.indent('func New() *{}'.format(self.class_name)):
```

Comment:
The function name "New" should be generated together with the class, so that different variants can be in the same folder. As an example NewLohnsteuer2020 or NewLohnsteuer2018 etc.